### PR TITLE
Fix `BitButton` and friends to be type="submit" in a form via default

### LINKED ISSUE #2874: `BitButton` should trigger form onsubmit when inside a form
### Discussed in https://github.com/bitfoundation/bitplatform/discussions/2866

<div type='discussions-op-text'>

<sup>Originally posted by **linkdotnet** August  3, 2022</sup>
I am not sure if this is really feasible and I do understand that the idea is arguable at best. But anyway:

Any button, which doesn't define explicitly the `type="button"`, will be of `type="submit"`. 

> The missing value default and invalid value default are the Submit Button state.

Specification can be found here: https://html.spec.whatwg.org/multipage/form-elements.html#attr-button-type

That means if we have a component like this:
```razor
<EditForm Model="@MyModel" OnValidSubmit="HandleSubmit">
    <BitButton>Submit</BitButton>
</EditForm>

@code {
    private void HandleSubmit()
    {
        // Something to be invoked here
    }
}
```

Clicking on the button would invoke `HandleSubmit`. Only explicitly setting it to `ButtonType="BitButtonType.Submit"`. 
Now I do understand that the default is `ButtonType.Button` which translates to `type="button"`. The confusion comes from the point that if you would do that in html (without specifing the type) the form would be submitted.

One possible solution would be to make `ButtonType` optional aka nullable and check whether or not a `EditContext` is available.</div>

### PR DESCRIPTION
**Pull request by linkdotnet**
this closes #2874 

This PR addressed the issue, which was in the discussion #2866 .

I took the liberty and created a small PR, which might go in the right direction. Feedback very welcome.

### CODE REVIEW

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs [hunk 1]
```diff
@@ -156,7 +157,7 @@ public void BitButtonAriaHiddenTest(bool ariaHidden, bool expectedResult)
 
         var bitButton = com.Find(".bit-btn");
 
-        Assert.AreEqual(bitButton.HasAttribute("aria-hidden"), expectedResult);
+        Assert.AreEqual(expectedResult, bitButton.HasAttribute("aria-hidden"));
```
Comment (linkdotnet) [inline src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs:160]: Normally expected comes first then actual.

I can also go through the tests and align this everywhere.

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs [hunk 2]
```diff
@@ -174,6 +175,33 @@ public void BitButtonTypeOfButtonTest(BitButtonType buttonType)
...
         var buttonTypeName = buttonType == BitButtonType.Button ? "button" : buttonType == BitButtonType.Submit ? "submit" : "reset";
-        Assert.AreEqual(bitButton.GetAttribute("type"), buttonTypeName);
+        Assert.AreEqual(buttonTypeName, bitButton.GetAttribute("type"));
     }
-}
+
+    [TestMethod]
+    public void BitButtonSubmitStateInEditContextTest()
+    {
+        var com = RenderComponent<BitButton>(parameters =>
```
Comment (linkdotnet) [inline src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs:184]: I explicitly did not test against `BitButtonTest` as directly testing is way easier.

In theory (in a follow up PR) we could remove this `BitButtonTest` and friends completely. The only reason is to have a `CascadingParameter` which we could add via `p => p.AddCascadingValue` anyway.

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs (hunk 1 above)
Comment (msynk) [reply src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs:160]: tnx, that would be great 🚀

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs (hunk 2 above)
Comment (msynk) [reply src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs:184]: yeah, you are right about these ...Test.razor component completely.
we kept adding these components as a good practice for future requirements (if any come up), but for now, we don't have any specific use for them.
although these middleware components are creating some issues/bugs testing 2-way parameters. let's discuss it in another thread.

File: src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs (hunk 1 above)
Comment (linkdotnet) [reply src/BlazorUI/Bit.BlazorUI.Tests/Buttons/BitButtonTests.cs:160]: Said and done.

**Review by msynk — APPROVED** [review]
tnx a lot, great job 🚀🚀🚀

**Comment by linkdotnet** [comment]
> tnx a lot, great job 🚀🚀🚀

Thanks as well for the super good support. I will leave it up to you guys whether or not you want to merge this. Or better when ;)

**Review by ysmoradi — APPROVED** [review]

**Comment by msynk** [comment]
> > tnx a lot, great job 🚀🚀🚀
> 
> Thanks as well for the super good support. I will leave it up to you guys whether or not you want to merge this. Or better when ;)

oh ofc, we are going to merge it (squash it here😉)

**Pull request merged**